moldesign/molecules/atomcollections.py

```diff
@@ -413,27 +435,86 @@ def transform(self, matrix):
+           atoms at ``atomlist.atoms``
     """
+    def __init__(self, atomlist=()):
+        atoms = []
+        for obj in atomlist:
+            if hasattr(obj, 'atoms'):
+                atoms.extend(obj.atoms)
+            else:
+                atoms.append(obj)
+        super(AtomList, self).__init__(atoms)
 
-    append = mdt.utils.Alias('atoms.append')
-    pop = mdt.utils.Alias('atoms.pop')
-    insert = mdt.utils.Alias('atoms.insert')
-    sort = mdt.utils.Alias('atoms.sort')
+    def __getitem__(self, item):
+        result = super(AtomList, self).__getitem__(item)
+        if isinstance(item, slice):
+            return type(self)(result)
+        else:
+            return result
 
-    def __init__(self, atomlist):
-        self.atoms = list(atomlist)
-        super(AtomList, self).__init__()
+    def __getslice__(self, i, j):
+        result = super(AtomList, self).__getslice__(i, j)
+        return type(self)(result)
 
-    def __iter__(self):
-        return iter(self.atoms)
+    def __str__(self):
+        return '[Atoms: %s]' % ', '.join(atom._shortstr() for atom in self)
 
-    def __getitem__(self, item):
-        return self.atoms[item]
+    def __repr__(self):
+        try:
+            return '<AtomList: [%s]>' % ', '.join(atom._shortstr() for atom in self)
+        except:
+            return '<AtomList at %x (__repr__ failed)>' % id(self)
 
+    def intersection(self, *otherlists):
+        """ Return a list of atoms that appear in all lists (including this one).
+
+        Args:
+            *otheriters (Iterable): one or more lists of atoms
+
+        Returns:
+            moldesign.AtomList: intersection of this lists with all passed lists. Preserves order
+              in this list
+        """
+        s = set(self).intersection(*otherlists)
+        return type(self)(o for o in self if o in s)
+
+    def union(self, *otherlists):
+        """ Return a list of atoms that appear in any lists (including this one).
+
+        Args:
+            *otherlists (Iterable): one or more lists of atoms
+
+        Returns:
+            moldesign.AtomList: union of this list of atoms with all passed lists of atoms.
+               Equivalent to concatenating all lists then removing duplicates
+        """
+        found = set()
+        newlist = type(self)()
+        for item in itertools.chain(self, *otherlists):
+            if item not in found:
+                found.add(item)
+                newlist.append(item)
+        return newlist
+
+    def unique(self):
+        """ Return only unique atoms from this list
+
+        Returns:
+            moldesign.AtomList: copy of this list without any duplicates. Preserves order.
+        """
+        return self.union()
+
+    def __sub__(self, other):
+        otherset = set(other)
+        return type(self)(atom for atom in self if atom not in otherset)
+
+    # alas for self so that this works with AtomContainer methods
```

Comment:
I started reading this as a really deep line of poetry until I realized you meant "alias" :)